### PR TITLE
clean operator crd after alert is deleted

### PR DESCRIPTION
**Pull request by aiwantaozi**
1. clean operator crd after alert is deleted
2. use sequence map key
3. update metric expression

### CODE REVIEW

File: pkg/controllers/user/alert/configsyncer/configsyncer.go [hunk 1]
```diff
@@ -109,104 +106,53 @@ func (d *ConfigSyncer) sync() error {
...
-				return errors.Wrapf(err, "get prometheus rule %s failed", d.clusterName)
-			}
-		}
-
-		updated := old.DeepCopy()
-		updated.Spec = promRule.Spec
-		if _, err = d.operatorCRDManager.PrometheusRules.Update(updated); err != nil {
-			return errors.Wrapf(err, "update prometheus rule %s failed", d.clusterName)
-		}
+	for k := range pAlertsMap {
```
Comment (orangedeng) [inline pkg/controllers/user/alert/configsyncer/configsyncer.go]: same as `cAlertsKey` list, can be generated in the loop above.

**Comment by cjellick** [comment]
In the future, I need better code commit messages that have a short descriptive subject and then explain the problem being address and how its being fixed.

Here is an example of a good commit message:
https://github.com/rancher/rancher/commit/43df127af8044a388e600c5e40a533c7f5f1c17c

Additionally, youve grouped three separate changes inot a single commit. They should at the very least be separate commits, but in this case they should probably be separate PRs as they appear to be very distinct different issues.


You don't need to change it for this PR, but please take this into account in the future.

File: pkg/controllers/user/alert/configsyncer/configsyncer.go [hunk 2]
```diff
@@ -109,104 +109,53 @@ func (d *ConfigSyncer) sync() error {
...
 	for _, alert := range clusterAlertRules {
 		if alert.Status.AlertState != "inactive" {
-			cAlertsAlertmanagerMap[alert.Spec.GroupName] = append(cAlertsAlertmanagerMap[alert.Spec.GroupName], alert)
-
-			if alert.Spec.MetricRule != nil {
-				haveClusterMetrics = true
-
-				cAlertsOperatorMap[alert.Spec.GroupName] = append(cAlertsOperatorMap[alert.Spec.GroupName], alert)
-			}
+			cAlertsMap[alert.Spec.GroupName] = append(cAlertsMap[alert.Spec.GroupName], alert)
```
[the remark this answers is not in the document]
Comment (aiwantaozi) [reply pkg/controllers/user/alert/configsyncer/configsyncer.go]: I need a key array include the groupName, if I just append, will have duplicated element

[the remark this answers is not in the document]
Comment (aiwantaozi) [reply pkg/controllers/user/alert/configsyncer/configsyncer.go]: I need a key array include the groupName, if I just append, will have duplicated element

**Comment by cjellick** [comment]
@orangedeng LGTMd outside of here. going to merge

**Comment by aiwantaozi** [comment]
@cjellick Thanks, I will improve the PR next time.

**Pull request merged**